src/gui/layout/qgslayoutatlaswidget.cpp

```diff
@@ -240,7 +240,11 @@ void QgsLayoutAtlasWidget::changesSortFeatureExpression( const QString &expressi
 
   mBlockUpdates = true;
   mLayout->undoStack()->beginCommand( mAtlas, tr( "Change Atlas Sort" ) );
-  mAtlas->setSortExpression( expression );
+  QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( mAtlasCoverageLayerComboBox->currentLayer() );
+  if ( QgsExpression::expressionToLayerFieldIndex( expression, vlayer ) != -1 && !expression.contains( '\"' ) )
```

Comment:
@roya0045 given that we're likely to need this fix in many different places (eg https://github.com/qgis/QGIS/pull/46546 ), could we extract this logic out to a `static QString QgsExpression::quoteExpressionIfNeeded( const QString& expression, QgsVectorLayer* layer = nullptr )` method?

Reply:
I have no issues doing it, should be done in the coming days.

Reply:
Implemented, did the tests too, though I was unsure if leaving the whitespaces at the start and end is a good choice. Are whitespaces in fields critical? I would propose calling 'trimmed' on the string, either in the columnrefquote of before passing the string to it. 

What's your stance on trimming the strings?